### PR TITLE
[CHE-172] Refactor Server Startup and DB Connection

### PR DESCRIPTION
**Pull request by seantokuzo**
### Description

As part of the CHE-167 story, this PR:
- separates app configuration logic from server startup logic
- adds environment variable checks to server startup function
- adds necessary environment variable to docker compose files for development and testing
- simplifies imports of routers
- fixes imports and logic in existing tests in `db.test.ts` and `index.test.ts`
- comments out tests that need refactoring in `db.test.ts`

### Jira Task

[JIRA TICKET](https://codehammers.atlassian.net/browse/CHE-172?atlOrigin=eyJpIjoiZWUxNzcwM2IwNDQ1NGM1NmFhZDE4MTcyODhiYmVkNjkiLCJwIjoiaiJ9)

### Testing Instructions
- run npm run docker-remove-all to remove local containers/images/volumes
- run docker build -f Dockerfile-dev -t codehammers/ch-dev-dep-v2 . to build the dev image including the express-async-errors package
- run npm run docker-build-check to run linting and testing

#### All Team Members

- [x] I added a descriptive title to this PR.
- [x] I filled out the **Description**, **Jira Task**, and **Testing Instructions** sections above.
- [x] I added or updated [Jest unit tests]for any changes to components, server-side controllers, etc.
- [x] I ran `npm run docker-build-check` in my local environment to check that this PR passes all linting checks and unit tests.
- [x] I did a quick check to make sure my code changes follow the recommended style guide.

### CODE REVIEW

File: docker-compose-dev.yml [hunk 1]
```diff
@@ -13,6 +13,15 @@ services:
     command: npm run dev-ts
     environment:
       - NODE_ENV=development
+      - POSTGRES_USER=postgres
+      - POSTGRES_DB=ch-dev-database
+      - POSTGRES_PASSWORD=ch-dev
+      - AWS_ACCESS_KEY_ID=placeholder-value
+      - AWS_SECRET_ACCESS_KEY=placeholder-value
```
Comment (brok3turtl3) [inline docker-compose-dev.yml:20]: At this point, we should probably just move all the env's to here. It makes onboarding easier and they're probably even a little more secure in a file nobody wants to look at lol...

File: docker-compose-dev.yml [hunk 2]
```diff
@@ -13,6 +13,15 @@ services:
...
+      - AWS_REGION=placeholder-value
+      - BUCKET_NAME=placeholder-value
+      # suppress aws sdk v2 deprecation warning
+      - AWS_SDK_JS_SUPPRESS_MAINTENANCE_MODE_MESSAGE=1;
```
Comment (brok3turtl3) [inline docker-compose-dev.yml:24]: LOL - Out of sight out of mind? As annoying as it is, 
I'm not sure this should fall off our radar.

File: docker-compose-test.yml [hunk 3]
```diff
@@ -13,8 +13,17 @@ services:
     depends_on:
       - mongo
     environment:
-      - MONGO_URI=mongodb://mongo:27017/ch-testdb
       - JWT_SECRET=${JWT_SECRET}
+      - MONGO_URI=mongodb://mongo:27017/ch-testdb
+      - POSTGRES_USER=postgres
+      - POSTGRES_DB=ch-dev-database
+      - POSTGRES_PASSWORD=ch-dev
+      - AWS_ACCESS_KEY_ID=placeholder-value
```
Comment (brok3turtl3) [inline docker-compose-test.yml:21]: Same as above.

File: server/config/db.ts [hunk 4]
```diff
@@ -1,24 +1,15 @@
...
   try {
-    // Check that MONGO_URI is defined
-    if (!process.env.MONGO_URI) {
-      throw new Error('MONGO_URI must be defined in the environment variables.');
-    }
-
-    const connection = await mongoose.connect(process.env.MONGO_URI);
-
-    console.log(`MongoDB is connected to: ${connection.connection.host}`);
+    const connection = await mongoose.connect(mongoUri);
```
Comment (brok3turtl3) [inline server/config/db.ts:6]: Great way to make the connection flexible.

File: server/config/db.ts [hunk 5]
```diff
@@ -1,24 +1,15 @@
...
+    console.log(`🍃 MongoDB is connected to: ${connection.connection.host}`);
   } catch (error) {
     if (error instanceof Error) {
       console.error(error.message);
-    } else {
-      console.error('❌ Error connecting to the database ❌');
     }
-    process.exit(1);
+    throw new DatabaseConnectionError();
   }
```
Comment (brok3turtl3) [inline server/config/db.ts:13]: Love seeing the new custom errors being used.

File: server/index.ts [hunk 6]
```diff
@@ -1,70 +1,32 @@
...
+  if (!process.env.AWS_REGION) throw Error('❌ AWS_REGION must be defined!');
+  if (!process.env.BUCKET_NAME) throw Error('❌ BUCKET_NAME must be defined!');
+
+  // Connect to MongoDB
+  await connectDB(process.env.MONGO_URI);
+
+  // Startup the server
   return app.listen(PORT, () =>
-    console.log(`Server running in ${process.env.NODE_ENV} mode on port ${PORT}`),
+    console.log(`💥 ${hazzah}Server running in ${process.env.NODE_ENV} mode on port ${PORT}`),
```
Comment (brok3turtl3) [inline server/index.ts:26]: Thank you 😍

**Review by brok3turtl3 — CHANGES_REQUESTED** [review]
Great re-org here @seantokuzo ! 👍  It was overdue!
 
I think the front and back testing files need to be reviewed at this point and a fresh slate of refactoring and new testing tickets put up. Let's chat about that this week.

I left mostly attaboys but had a few questions around the compose files. ❓

File: docker-compose-dev.yml (hunk 1 above)
Comment (seantokuzo) [reply docker-compose-dev.yml:20]: I'll make that part of the mongo dev container ticket. Don't wanna push your atlas uri... or do I? 👹

File: docker-compose-dev.yml (hunk 2 above)
Comment (seantokuzo) [reply docker-compose-dev.yml:24]: We have til September lol. I think it'll still work when it goes into maintenance mode

File: docker-compose-dev.yml (hunk 1 above)
Comment (brok3turtl3) [reply docker-compose-dev.yml:20]: Forgot about that - we already talked about it lol.. Consider this resolved! =)

**Review by brok3turtl3 — APPROVED** [review]
🎉  Nice work @seantokuzo!

File: server/index.ts (hunk 6 above)
Comment (seantokuzo) [reply server/index.ts:26]: Hazzah!

File: server/config/db.ts (hunk 5 above)
Comment (seantokuzo) [reply server/config/db.ts:13]: if only that db test wasn't spying on process.exit lol

File: server/config/db.ts (hunk 4 above)
Comment (seantokuzo) [reply server/config/db.ts:6]: for when we start sharding hard

**Pull request merged**